ciri/conf/provider.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __CIRI_CONF_PROVIDER_H__
+#define __CIRI_CONF_PROVIDER_H__
+
+#include <getopt.h>
```

Comment:
This should be in `ciri/conf/getopt.c` as `ciri/conf/provider.h` is supposed to be abstract